### ISSUE TITLE
Running tests causes Segfault

### ISSUE THREAD

**Issue by leventyalcin** [opsgang/libs#9]
Well, I'm going to write this down in here just to remember. 

I was trying to run tests locally while I was adding a feature. If I try to run any of the tests this happens

```
bash-4.4$ ./t/terraform/terraform_dev.functions t_tf_init_module
+ ./t/terraform/terraform_dev.functions t_tf_init_module
INFO ./t/terraform/terraform_dev.functions: sourcing ./t/terraform/terraform_dev.functions
INFO ./t/terraform/terraform_dev.functions: sourcing ./t/terraform/terraform_dev.functions
.
.
. many more
Segmentation fault: 11
```

```
bash-4.4$ ./t/terraform/terraform_run.functions
INFO ./t/terraform/terraform_run.functions: sourcing ./t/terraform/terraform_run.functions
INFO ./t/terraform/terraform_run.functions: sourcing ./t/terraform/terraform_run.functions
INFO ./t/terraform/terraform_run.functions: sourcing ./t/terraform/terraform_run.functions
.
. many more
Segmentation fault: 11
```
My finding is this https://github.com/opsgang/libs/blob/master/bash/t/terraform/terraform_run.functions#L365

The test runner file calls `source_src_and_deps` and the function tries to source given file names and also [`$SRC`](https://github.com/opsgang/libs/blob/master/bash/t/t.functions#L31) which is the file itself. So, that causes the infinitive loop. 

We need to break that single file down to two files, a runner and functions. With that way, we can resolve the self sourcing issue. 

Interestingly, I have no clue how that works on CI 😕

**Comment by jinal--shah** [comment]
Works absolutely fine locally for me on both macos and inside containers, and on coreos.

Also see this: https://github.com/opsgang/libs/blob/master/bash/t/t.functions#L3

$SRC is  not the file itself (the test script) but the corresponding .functions file e.g if you are running
t/terraform/terraform_dev.functions, $SRC is terraform/terraform_dev.functions.

**Comment by jinal--shah** [comment]
```
*shahj@ELSOXFM-213982* 11:28:56 bash [git:tf-init-module] $ t/terraform/terraform_dev.functions
INFO t/terraform/terraform_dev.functions: sourcing terraform/terraform_dev.functions
INFO t/terraform/terraform_dev.functions: sourcing habitual/std.functions
INFO t/terraform/terraform_dev.functions: sourcing habitual/git.functions
INFO t/terraform/terraform_dev.functions: running test for: tf_init_module
INFO t/terraform/terraform_dev.functions:     running test for: [tf_init_module()] initialise_empty_module
INFO terraform_dev.functions:tf_init_module(): Module terraform-aws-test-module initialised
INFO: ALL TESTS SUCCESSFUL
```

**Comment by jinal--shah** [comment]
I suspect your mac setup is wrong somehow?

**Comment by leventyalcin** [comment]
Raised #11 and Shippable tests seem ok

**Comment by jinal--shah** [comment]
Fixed by 78d1aec1e7db0176cf84a478d3c0293c53e24617
